docs/ui/components/Form/Checkbox.tsx

```diff
@@ -22,6 +22,7 @@ export const Checkbox = forwardRef(function Checkbox(
         id={id}
         ref={ref}
         checked={checked}
+        onChange={onChange}
         data-label={label}
         disabled={disabled}
         className={mergeClasses(
```

Comment:
I have tested the fix locally, and it still does not work as expected, due to how we construct Checkbox element. 

What the solution misses is `pointer-events-none` class on `CheckIcon`, otherwise, you are able to select checkbox, but not deselect it by clicking on the box with check.